File: scdproiect/src/main/java/com/utcn/scdproiect/Courier/Courier.java
package com.utcn.scdproiect.Courier;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Builder
@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Courier {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String name;

    @Column(unique = true)
    private String email;

    private String password;

    @Builder.Default
    @Column()
    private boolean isManager=false;

    // Change manager field to an Integer for manager_id
    @Column(name = "manager_id")
    private Integer managerId;

    // Optional: List of subordinates for a given manager
    @OneToMany(mappedBy = "managerId", cascade = CascadeType.ALL)
    private List<Courier> subordinates;
}

File: CourierAppWindowsForm/Courier.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CourierAppClient
{
    internal class Courier
    {
       public string name { get; set; } 

       public int id { get; set; }

       public string email { get; set; }

        public int managerId = 0;

       public bool isManager { get; set; }

       public List<Courier> subordinates { get; set; }
    }
}
 